test/test_sync.py

```diff
@@ -91,65 +128,281 @@ def test_two(self):
+    Makes a File object for a b2 file, with one FileVersion for
+    each modification time given in *args.
 
-        folder_a = FakeFolder('local', [file_a1, file_a2, file_a3, file_a4])
-        folder_b = FakeFolder('b2', [file_b1, file_b2, file_b3, file_b4])
+    Positive modification times are uploads, and negative modification
+    times are hides.  It's a hack, but it works.
 
-        actions = list(make_folder_sync_actions(folder_a, folder_b, 1))
-        self.assertEqual(
+        b2_file('a.txt', 300, -200, 100)
+
+    Is the same as:
+
+        File(
+            'a.txt',
             [
-                "b2_upload(/dir/a.txt, a.txt, 100)", "b2_delete(b.txt, id_b_200)",
-                "b2_upload(/dir/e.txt, e.txt, 300)"
-            ], list(map(str, actions))
+               FileVersion('id_a_300', 'a.txt', 300, 'upload'),
+               FileVersion('id_a_200', 'a.txt', 200, 'hide'),
+               FileVersion('id_a_100', 'a.txt', 100, 'upload')
+            ]
         )
+    """
+    versions = [
+        FileVersion(
+            'id_%s_%d' % (name[0], abs(mod_time)), 'folder/' + name, abs(mod_time), 'upload'
+            if 0 < mod_time else 'hide', 10
+        ) for mod_time in args
+    ]  # yapf disable
+    return File(name, versions)
 
-    def test_b2_to_local(self):
-        file_a1 = File("a.txt", [FileVersion("id_a_100", 100, "upload")])  # only in source
-        file_a2 = File("c.txt", [FileVersion("id_c_200", 200, "upload")])  # mod time matches
-        file_a3 = File("d.txt", [FileVersion("id_d_100", 100, "upload")])  # newer in dest
-        file_a4 = File("e.txt", [FileVersion("id_e_300", 300, "upload")])  # newer in source
 
-        file_b1 = File("b.txt", [FileVersion("/dir/b.txt", 200, "upload")])  # only in dest
-        file_b2 = File("c.txt", [FileVersion("/dir/c.txt", 200, "upload")])  # mod time matches
-        file_b3 = File("d.txt", [FileVersion("/dir/d.txt", 200, "upload")])  # newer in dest
-        file_b4 = File("e.txt", [FileVersion("/dir/e.txt", 200, "upload")])  # newer in source
+def local_file(name, *args):
+    """
+    Makes a File object for a b2 file, with one FileVersion for
+    each modification time given in *args.
+    """
+    versions = [FileVersion('/dir/%s' % (name,), name, mod_time, 'upload', 10) for mod_time in args]
+    return File(name, versions)
 
-        folder_a = FakeFolder('b2', [file_a1, file_a2, file_a3, file_a4])
-        folder_b = FakeFolder('local', [file_b1, file_b2, file_b3, file_b4])
 
-        actions = list(make_folder_sync_actions(folder_a, folder_b, 1))
-        self.assertEqual(
-            [
-                "b2_download(a.txt, id_a_100)", "local_delete(/dir/b.txt)",
-                "b2_download(e.txt, id_e_300)"
-            ], list(map(str, actions))
+class TestMakeSyncActions(unittest.TestCase):
+    def setUp(self):
+        self.reporter = MagicMock()
+
+    def test_illegal_b2_to_b2(self):
+        b2_folder = FakeFolder('b2', [])
+        try:
```

Comment:
then we should implement the wrapper by ourselves in one place instead of writing it in each test